### DISCUSSION TITLE
Where does NextJS store statically generated pages?

### DISCUSSION THREAD

**Discussion by ApplePieGiraffe** [vercel/next.js#22117] [Help] [answered]
I read in the official NextJS tutorial that NextJS generates HTML files from my pages at build time and stores them so that they can be sent immediately to clients when they are requested for (static generation).

Where does NextJS store these completed HTML pages? Is there any way I can find them to see what they look like?

**Comment by nicosh** [comment, marked as answer]
I think pages are stored inside `./next/server/pages`

**Comment by ats1999** [comment]
Is there any way we can change it?
I would like to store it on the cdn

**Reply by Colbitron** [reply]
You should still be able to access those files via CDN once you have uploaded them to a server that you control. This could even be something like Amazon S3 where it is relatively cheap to host the files there and use a CDN to access, cache, and serve those files.
